src/refmt_impl.ml

```diff
@@ -35,19 +37,34 @@ let reasonBinaryParser use_stdin filename =
   let (magic_number, filename, ast, comments, parsedAsML, parsedAsInterface) = input_value chan in
   ((ast, comments), parsedAsML, parsedAsInterface)
 
-let ocamlBinaryParser use_stdin filename parsedAsInterface =
-  let chan =
-    match use_stdin with
-      | true -> stdin
-      | false ->
-          let file_chan = open_in filename in
-          seek_in file_chan 0;
-          file_chan
+let ocamlBinaryParser use_stdin filename =
+  let chan = match use_stdin with
+    | true -> stdin
+    | false ->
+      let file_chan = open_in filename in
+      seek_in file_chan 0;
+      file_chan
   in
-  let _ = really_input_string chan (String.length Config.ast_impl_magic_number) in
-  let _ = input_value chan in
-  let ast = input_value chan in
-  ((ast, []), true, parsedAsInterface)
+  match Ast_io.from_channel chan with
```

Comment:
No it doesn't. The most reasonable to me would be to keep ocaml implementations of Marshal compatible with each versions.
For now, Marshal didn't break since 4.02.